### DISCUSSION TITLE
consider increasing default podman service timeout?

### DISCUSSION THREAD

**Discussion by nivekuil** [containers/podman#11581] [General]
The default timeout for `podman system service` is 5 seconds. For my use case where I was polling the podman socket for health checks every 10 seconds, restarting the service process so often ended up having a substantial cpu overhead. On one machine the idle cpu dropped 10 percentage points (/100) when I bumped -t to 300 seconds. Would it make sense to bump this default timeout to something a bit higher than 5s? I think it may generally cost more resources than it saves with such a low timeout.

**Comment by rhatdan** [comment]
How about if we make it easier to customize.
https://github.com/containers/common/pull/772

**Reply by Luap99** [reply]
I think the question is not so much about how to change it, it is more about why the default is 5.

> I think it may generally cost more resources than it saves with such a low timeout.

**Reply by rhatdan** [reply]
It used to be longer, and then it never went away.  For most use cases it works fine if it closes 5 seconds after the last connection.  Setting it to something very long, just makes it a daemon and means that it probably never closes.

**Reply by nivekuil** [reply]
Thanks for adding the config option. That's much easier to tweak than the systemd unit.

If the service is constantly receiving connections, then isn't never closing and effectively being a daemon the optimal behavior? The exact interval would be a tradeoff but I do think 5 seconds is a bit short for many common use cases people have for the docker socket.
